packages/app-storage/src/Query.tsx

```diff
@@ -24,35 +26,67 @@ type ComponentProps = {};
 
 type State = {
   inputs: Array<any>, // node?
-  Component: React.ComponentType<ComponentProps>;
+  Component: React.ComponentType<ComponentProps>
 };
 
-const cache: Array<React.ComponentType<ComponentProps>> = [];
+type CacheInstance = {
+  component: React.ComponentType<any>,
+  render: RenderFn,
+  refresh: (swallowErrors: boolean, contentShorten: boolean) => React.ComponentType<any>
+};
+
+const cache: Array<CacheInstance> = [];
 
 class Query extends React.PureComponent<Props, State> {
   state: State = {} as State;
 
-  static getCachedComponent (query: QueryTypes): React.ComponentType<ComponentProps> {
+  static getCachedComponent (query: QueryTypes): CacheInstance {
     const { id, key, params = [] } = query as StorageModuleQuery;
 
     if (!cache[id]) {
       const values: Array<any> = params.map(({ value }) => value);
       const type = key.meta
         ? key.meta.type.toString()
         : 'Data';
-
-      cache[query.id] = withObservableDiv('rawStorage', { params: [key, ...values] })(
-        (value: any) =>
-          valueToText(type, value),
-        { className: 'ui--output' }
+      const defaultProps = { className: 'ui--output' };
+
+      /* render function to create an element for the query results which is plugged to the api */
```

Comment:
(insubstantial nitpick) Can we change this to be `//` comments - it just used everywhere else. (There are a number of these)